components/PlantForm/PlantForm.js

```diff
@@ -19,20 +20,36 @@ export default function PlantForm({ onSubmit, isEdit, editPlant }) {
     );
   }
 
+  const defaultImage =
+    "https://images.pexels.com/photos/29780009/pexels-photo-29780009/free-photo-of-entspannte-orange-und-weisse-katze-sonnt-sich-im-sonnenlicht.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2";
+
   async function handleSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
+    let imageUrl;
+
+    if (data.imageUrl.size > 0) {
+      const response = await fetch("/api/upload", {
+        method: "POST",
+        body: formData,
+      });
+
+      if (!response.ok) {
+        console.error("Whoopsie");
+        return;
+      }
+      let { url } = await response.json();
+      imageUrl = url;
+    }
     const newPlant = {
       name: data.name,
       botanicalName: data.botanicalName,
       lightNeed: data.lightNeed,
       waterNeed: data.waterNeed,
       fertiliserSeason: fertiliserSeason,
       description: data.description,
-      imageUrl: isEdit
-        ? editPlant?.imageUrl
-        : "https://images.pexels.com/photos/2587313/pexels-photo-2587313.jpeg?auto=compress&cs=tinysrgb&w=1600",
+      imageUrl: imageUrl || editPlant?.imageUrl || defaultImage,
```

Comment:
Maybe you can think about here of using a state instead of 3 different variables